TCD1304.py

```diff
@@ -62,29 +58,32 @@ def power_source ():        # puts a voltage of 4V on PV1
         ps.pv1 = 4
         ps.pv1
     
+    def analog_signal_read (self, block:False):
+        self.scope = Oscilloscope()
+        self.scope.select_range('CH1', max_voltage_output)   # voltagerange should be fitted to the sensors output for better resolution. sensor otput is between 2V-3V due to datasheet
+        self.scope.configure_trigger(channel = 'CH1', voltage = min_voltage_output ) # starts recording , when voltage is over a certain level
+        self.scope.capture(channels=1, samples=integration_elements*samples_per_element, timegap=integration_time/samples_per_element)
+        y1, y2 = self.scope.fetch_data()
+        diff = abs(y1,y2[1, 0] - min_voltage_output )
+        analog_measurement = np.ndarray ()
+        return analog_measurement  # this should put the Measurements and timestamps in an array
+    
+    
     def master_clock (self):        # puts PWM with frequency of 2MHz on SQ1; masterclock is the fastest clock
        prescaler = int(math.log(oscillator_frequency / frequency_master_clock) / math.log(2))   # When setting a frequency by mapping the reference clock directly to a PWM output, only frequencies which are even factors of 128 MHz (the frequency of the PSLab's main oscillator) are available. The frequency is therefore not set by specifying the frequency itself, but by setting a prescaler.
-        self.pwmgen.map_reference_clock(["SQ1"], prescaler)    
+       self.pwmgen.map_reference_clock(["SQ1"], prescaler)    
     
-    def sh_clock (self):           #  on SQ2 sets pulse for the integrationtime. Running the Sensor in shutter mode try with a frequency of 1/4 of master_clock. tint (min) = 10µs
-        self.pwmgen.generate(["SQ2"], frequency_master_clock/4, [0.5] ) 
+    def sh_clock (self):           #  on SQ2 sets pulse for the integrationtime. Running the Sensor in shutter mode try with tint (min) = 10µs
+        self.pwmgen.generate(["SQ2"], 1/integration_time, [0.5] ) 
 
 
     def icg_clock (self):           #  on SQ3 starts and stops the reading of the sensor; this third clock is the slowest clock

```

Comment:
Looking at this method again, I think it is not correct. What you want to do is set ICG high for the duration of `readout_time`, which would just be:
```python
self.pwmgen.set_state(sq3=True)
time.sleep(readout_time)
self.pwmgen.set_state(sq3=False)
```
I think I was looking at the SH clock before when I suggested the for-loop implementation. Sorry for the confusion.